### PR TITLE
Basic infinite runner

### PR DESCRIPTION
**Pull request by tigleym**

### CODE REVIEW

File: src/runner/backgroundlayer.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+use bevy::prelude::*;
+
+#[derive(Debug, Component)]
+struct BackgroundLayer;
+
+pub struct BackgroundLayerPlugin;
+
+impl Plugin for BackgroundLayerPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_startup_system(spawn_background.system())
```
Comment (NightlySide) [inline src/runner/backgroundlayer.rs:11]: Remove the `.system()` it has been deprecated in 0.6

File: src/runner/backgroundlayer.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                ..Default::default()
+            },
+            ..Default::default()
+        })
+        .insert(BackgroundLayer);
+}
+
+fn move_background(
+    _camera: Query<&Transform, With<Player>>,
+    mut _background: Query<&mut Transform, (With<BackgroundLayer>, Without<Player>)>,
```
Comment (NightlySide) [inline src/runner/backgroundlayer.rs:52]: To verify but I think here parenthesis are optional

File: src/runner/player.rs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+struct AnimationTimer(Timer);
+
+pub struct PlayerPlugin;
+
+impl Plugin for PlayerPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_startup_system(spawn_character.system())
+            .add_system(move_character.system())
+            .add_system(follow_player_camera.system())
+            .add_system(animate_sprite.system());
```
Comment (NightlySide) [inline src/runner/player.rs:20]: Same here, remove the `.system()` suffix

File: src/runner/player.rs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            sprite.index = (sprite.index + 1) % texture_atlas.textures.len();
+        }
+    }
+}
+
+fn move_character(
+    keyboard_input: Res<Input<KeyCode>>,
+    mut query: Query<(&Player, &mut Transform, &mut Handle<TextureAtlas>)>,
+) {
+    for (player, mut transform, mut _atlas) in query.iter_mut() {
```
Comment (NightlySide) [inline src/runner/player.rs:71]: If you are not using the `_atlas` variable, the whole thing can be : 
```rust
for (player, mut transform, _) in query.iter_mut() {
```

**Review by NightlySide — CHANGES_REQUESTED** [review]

File: src/runner/backgroundlayer.rs (hunk 2 above)
Comment (tigleym) [reply src/runner/backgroundlayer.rs:52]: I get this error if I remove the parenthesis: 

```
error[E0107]: this struct takes at most 2 generic arguments but 3 generic arguments were supplied
   --> src\runner\backgroundlayer.rs:52:22
    |
52  |     mut _background: Query<&mut Transform, With<BackgroundLayer>, Without<Player>>,
    |                      ^^^^^ expected at most 2 generic arguments   --------------- help: remove this generic argument
    |
note: struct defined here, with at most 2 generic parameters: `Q`, `F`
```

Comment (cdsupina) [reply src/runner/backgroundlayer.rs:52]: I think you are correctly using With/Without, but we may want to consider instead just having a tag component for the background, or a core component if it needs any data managed.

File: src/runner/backgroundlayer.rs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+#[derive(Debug, Component)]
+struct BackgroundLayer;
+
+pub struct BackgroundLayerPlugin;
+
+impl Plugin for BackgroundLayerPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_startup_system(spawn_background)
+            .add_system(move_background.system());
```
Comment (cdsupina) [inline src/runner/backgroundlayer.rs:12]: This `.system()` can be removed too.

**Review by cdsupina — APPROVED** [review]

**Pull request merged**